public/src/components/campaign/campaign_list.jsx

```diff
@@ -25,11 +32,14 @@ export default class CampaignList extends Component {
             <TableListView
               item_collection={all_campaigns}
               account_info={account_info}
+              auth={auth}
               history={history}
               button_collection={account_info.is_admin ?
                                    admin_campaigns :
                                    volunteer_campaigns}
-              setCurrentItem={setCurrentCampaign}
+              setCurrentItem={account_info.is_admin ?
```

Comment:
If I'm not mistaken, this will make the user join a campaign if they just click on the campaign row itself. I'm not sure we want this to happen. We only want this action to be triggered when they click the Join button aka addCampaignToUser should only be called on the handleClick event for the button.

Reply:
I just tested this out. Clicking on the campaign row does not trigger any actions - on user view or admin view. Only clicking the button triggers the action. Andi had a suggestion below to add handleRedirectClick and handleActionClicks on the buttons. That might fix the confusion. Looking into it now.